File: LearningLens2025/legacy_files/focused-ai-grader-service/focusedai-platform/backend/src/main/java/com/focusedai/model/execution/CodeAnalysis.java
package com.focusedai.model.execution;

import java.util.List;
import java.util.Map;
import java.util.HashMap;
import java.util.ArrayList;

public class CodeAnalysis {
    private String language;
    private int fileCount;
    private boolean hasMainMethod;
    private boolean hasScanner;
    private boolean hasFileIO;
    private boolean hasTestAnnotations;
    private boolean hasSystemOut;
    private boolean isPackageExecution;
    private String packageName;
    private String mainClassName;
    private String recommendedStrategy;
    private String targetMethod;
    private List<String> classNames = new ArrayList<>();
    private List<String> publicMethods = new ArrayList<>();
    private List<String> detectedFeatures = new ArrayList<>();
    private double confidence;
    
    // Constructors
    public CodeAnalysis() {}
    
    // Convert to map for serialization
    public Map<String, Object> toMap() {
        Map<String, Object> map = new HashMap<>();
        map.put("language", language);
        map.put("fileCount", fileCount);
        map.put("hasMainMethod", hasMainMethod);
        map.put("hasScanner", hasScanner);
        map.put("hasFileIO", hasFileIO);
        map.put("hasTestAnnotations", hasTestAnnotations);
        map.put("hasSystemOut", hasSystemOut);
        map.put("isPackageExecution", isPackageExecution);
        map.put("packageName", packageName);
        map.put("mainClassName", mainClassName);
        map.put("recommendedStrategy", recommendedStrategy);
        map.put("targetMethod", targetMethod);
        map.put("classNames", classNames);
        map.put("publicMethods", publicMethods);
        map.put("detectedFeatures", detectedFeatures);
        map.put("confidence", confidence);
        return map;
    }
    
    // Getters and setters
    public String getLanguage() { return language; }
    public void setLanguage(String language) { this.language = language; }
    
    public int getFileCount() { return fileCount; }
    public void setFileCount(int fileCount) { this.fileCount = fileCount; }
    
    public boolean isHasMainMethod() { return hasMainMethod; }
    public void setHasMainMethod(boolean hasMainMethod) { this.hasMainMethod = hasMainMethod; }
    
    public boolean isHasScanner() { return hasScanner; }
    public void setHasScanner(boolean hasScanner) { this.hasScanner = hasScanner; }
    
    public boolean isHasFileIO() { return hasFileIO; }
    public void setHasFileIO(boolean hasFileIO) { this.hasFileIO = hasFileIO; }
    
    public boolean isHasTestAnnotations() { return hasTestAnnotations; }
    public void setHasTestAnnotations(boolean hasTestAnnotations) { this.hasTestAnnotations = hasTestAnnotations; }
    
    public boolean isHasSystemOut() { return hasSystemOut; }
    public void setHasSystemOut(boolean hasSystemOut) { this.hasSystemOut = hasSystemOut; }
    
    public boolean isPackageExecution() { return isPackageExecution; }
    public void setPackageExecution(boolean packageExecution) { isPackageExecution = packageExecution; }
    
    public String getPackageName() { return packageName; }
    public void setPackageName(String packageName) { this.packageName = packageName; }
    
    public String getMainClassName() { return mainClassName; }
    public void setMainClassName(String mainClassName) { this.mainClassName = mainClassName; }
    
    public String getRecommendedStrategy() { return recommendedStrategy; }
    public void setRecommendedStrategy(String recommendedStrategy) { this.recommendedStrategy = recommendedStrategy; }
    
    public String getTargetMethod() { return targetMethod; }
    public void setTargetMethod(String targetMethod) { this.targetMethod = targetMethod; }
    
    public List<String> getClassNames() { return classNames; }
    public void setClassNames(List<String> classNames) { this.classNames = classNames; }
    
    public List<String> getPublicMethods() { return publicMethods; }
    public void setPublicMethods(List<String> publicMethods) { this.publicMethods = publicMethods; }
    
    public List<String> getDetectedFeatures() { return detectedFeatures; }
    public void setDetectedFeatures(List<String> detectedFeatures) { this.detectedFeatures = detectedFeatures; }
    
    public double getConfidence() { return confidence; }
    public void setConfidence(double confidence) { this.confidence = confidence; }
    
    @Override
    public String toString() {
        return String.format("CodeAnalysis{language='%s', strategy='%s', confidence=%.1f%%, features=%s}", 
                language, recommendedStrategy, confidence, detectedFeatures);
    }
}

File: LearningLens2025/legacy_files/focused-ai-grader-service/focusedai-platform/frontend/lib/features/code_execution/models/code_analysis.dart
class CodeAnalysis {
  final String language;
  final int fileCount;
  final bool hasMainMethod;
  final bool hasScanner;
  final bool hasFileIO;
  final bool hasTestAnnotations;
  final bool hasSystemOut;
  final bool isPackageExecution;
  final String? packageName;
  final String? mainClassName;
  final String recommendedStrategy;
  final String? targetMethod;
  final List<String> classNames;
  final List<String> publicMethods;
  final List<String> detectedFeatures;
  final double confidence;

  CodeAnalysis({
    required this.language,
    required this.fileCount,
    required this.hasMainMethod,
    required this.hasScanner,
    required this.hasFileIO,
    required this.hasTestAnnotations,
    required this.hasSystemOut,
    required this.isPackageExecution,
    this.packageName,
    this.mainClassName,
    required this.recommendedStrategy,
    this.targetMethod,
    this.classNames = const [],
    this.publicMethods = const [],
    this.detectedFeatures = const [],
    required this.confidence,
  });

  factory CodeAnalysis.fromJson(Map<String, dynamic> json) {
    return CodeAnalysis(
      language: json['language'] ?? '',
      fileCount: json['fileCount'] ?? 0,
      hasMainMethod: json['hasMainMethod'] ?? false,
      hasScanner: json['hasScanner'] ?? false,
      hasFileIO: json['hasFileIO'] ?? false,
      hasTestAnnotations: json['hasTestAnnotations'] ?? false,
      hasSystemOut: json['hasSystemOut'] ?? false,
      isPackageExecution: json['isPackageExecution'] ?? false,
      packageName: json['packageName'],
      mainClassName: json['mainClassName'],
      recommendedStrategy: json['recommendedStrategy'] ?? 'STDIN_STDOUT',
      targetMethod: json['targetMethod'],
      classNames: List<String>.from(json['classNames'] ?? []),
      publicMethods: List<String>.from(json['publicMethods'] ?? []),
      detectedFeatures: List<String>.from(json['detectedFeatures'] ?? []),
      confidence: (json['confidence'] ?? 0.0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'language': language,
      'fileCount': fileCount,
      'hasMainMethod': hasMainMethod,
      'hasScanner': hasScanner,
      'hasFileIO': hasFileIO,
      'hasTestAnnotations': hasTestAnnotations,
      'hasSystemOut': hasSystemOut,
      'isPackageExecution': isPackageExecution,
      if (packageName != null) 'packageName': packageName,
      if (mainClassName != null) 'mainClassName': mainClassName,
      'recommendedStrategy': recommendedStrategy,
      if (targetMethod != null) 'targetMethod': targetMethod,
      'classNames': classNames,
      'publicMethods': publicMethods,
      'detectedFeatures': detectedFeatures,
      'confidence': confidence,
    };
  }

  CodeAnalysis copyWith({
    String? language,
    int? fileCount,
    bool? hasMainMethod,
    bool? hasScanner,
    bool? hasFileIO,
    bool? hasTestAnnotations,
    bool? hasSystemOut,
    bool? isPackageExecution,
    String? packageName,
    String? mainClassName,
    String? recommendedStrategy,
    String? targetMethod,
    List<String>? classNames,
    List<String>? publicMethods,
    List<String>? detectedFeatures,
    double? confidence,
  }) {
    return CodeAnalysis(
      language: language ?? this.language,
      fileCount: fileCount ?? this.fileCount,
      hasMainMethod: hasMainMethod ?? this.hasMainMethod,
      hasScanner: hasScanner ?? this.hasScanner,
      hasFileIO: hasFileIO ?? this.hasFileIO,
      hasTestAnnotations: hasTestAnnotations ?? this.hasTestAnnotations,
      hasSystemOut: hasSystemOut ?? this.hasSystemOut,
      isPackageExecution: isPackageExecution ?? this.isPackageExecution,
      packageName: packageName ?? this.packageName,
      mainClassName: mainClassName ?? this.mainClassName,
      recommendedStrategy: recommendedStrategy ?? this.recommendedStrategy,
      targetMethod: targetMethod ?? this.targetMethod,
      classNames: classNames ?? this.classNames,
      publicMethods: publicMethods ?? this.publicMethods,
      detectedFeatures: detectedFeatures ?? this.detectedFeatures,
      confidence: confidence ?? this.confidence,
    );
  }

  @override
  String toString() {
    return 'CodeAnalysis(language: $language, strategy: $recommendedStrategy, confidence: ${confidence.toStringAsFixed(1)}%, features: $detectedFeatures)';
  }
}